pipelines/controller/src/main/java/org/openmrs/analytics/HiveTableManager.java

```diff
@@ -76,26 +75,75 @@ public void createResourceTables(
-              resource,
-              timestamp,
-              THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
-              thriftServerParquetPath,
-              resource);
-      statement.execute(sql);
 
-      // Drop canonical table if exists.
-      sql = String.format("DROP TABLE IF EXISTS default.%s", resource);
-      statement.execute(sql);
+    String sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s_%s USING PARQUET LOCATION '%s/%s/%s'",
+            resource,
+            timestamp,
+            Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
+            thriftServerParquetPath,
+            resource);
+    createResourceTable(connection, sql);
+
+    // Drop canonical table if exists.
+    sql = String.format("DROP TABLE IF EXISTS default.%s", resource);
+    createResourceTable(connection, sql);
+
+    // Create canonical table with latest parquet files.
+    sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s USING PARQUET LOCATION '%s/%s/%s'",
+            resource,
+            Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
+            thriftServerParquetPath,
+            resource);
+    createResourceTable(connection, sql);
+  }
+
+  /**
+   * The method creates resource tables with names suffixed with given timestamp if not present. It
+   * excepts complete parquet files path.
+   *
+   * @param resource
+   * @param timestamp
+   * @param thriftServerParquetPath
+   * @throws SQLException
+   */
+  public void createResourceTable(String resource, String timestamp, String thriftServerParquetPath)
+      throws SQLException {
+
+    String sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s_%s USING PARQUET LOCATION '%s'",
+            resource, timestamp, thriftServerParquetPath);
+    try (Connection connection = DriverManager.getConnection(jdbcUrl)) {
+      createResourceTable(connection, sql);
+    }
+  }
+
+  /**
+   * This method creates canonical resource tables if not present. It excepts complete parquet files
+   * path.
+   *
+   * @param resource
+   * @param thriftServerParquetPath
+   * @throws SQLException
+   */
+  public void createResourceCanonicalTable(String resource, String thriftServerParquetPath)
+      throws SQLException {
 
-      // Create canonical table with latest parquet files.
-      sql =
-          String.format(
-              "CREATE TABLE IF NOT EXISTS default.%s USING PARQUET LOCATION '%s/%s/%s'",
-              resource, THRIFT_CONTAINER_PARQUET_PATH_PREFIX, thriftServerParquetPath, resource);
+    String sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s USING PARQUET LOCATION '%s'",
+            resource, thriftServerParquetPath);
+    try (Connection connection = DriverManager.getConnection(jdbcUrl)) {
+      createResourceTable(connection, sql);
+    }
+  }
+
+  private void createResourceTable(Connection connection, String sql) throws SQLException {
```

Comment:
Should this function be named `executeSql` or something like that instead?

Reply:
sure